pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -288,13 +304,15 @@ func (c *component) containers() []corev1.Container {
 			{Name: DikastesSyncVolumeName, MountPath: "/var/run/dikastes"},
 		}
 
-		if c.config.WAFEnabled {
+		if c.config.WAFEnabled || c.config.SidecarInjectionEnabled {
 			commandArgs = append(
 				commandArgs,
-				"--waf-enabled",
 				"--waf-log-file", filepath.Join(CalicologsVolumePath, "waf", "waf.log"),
```

Comment:
should `--waf-log-file` go under the `if c.config.WAFEnabled` section below? What aobut `--waf-ruleset-file`?

At face value these both seem like arguments that are only relevant when WAF is enabled.

Reply:
The `--waf-enabled` is telling dikastes that the per-host enforcement is enabled for WAF, needed for internal decisions on requests that can come even from sidecars or TPROXY iptables rules. So the `--waf-log-file` and `--waf-ruleset-file` are the configs that are required by both.

Reply:
Perfect.

Honestly, the renaming of WAFEnabled -> PerHostWAFEnabled makes this SO much easier to read for me now - this makes sense.

Reply:
> The --waf-enabled is telling dikastes that the per-host enforcement is enabled for WA

It is probably a good idea to rename this argument on Dikastes (or add a new one and deprecate this one)....

```
--per-host-waf-enabled
```

^ Would be more accurate.